test/unit/subnet/ipv6_test.rb

```diff
@@ -0,0 +1,149 @@
+    subnet = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8:0000::')
+    refute subnet.valid?
+  end
+
+  test "the name should be unique in the domain scope" do
+    first = FactoryGirl.create(:subnet_ipv6, :with_domains)
+    subnet = FactoryGirl.build(:subnet_ipv6, :name => first.name, :domains => first.domains)
+    refute subnet.valid?
+  end
+
+  test "when to_label is applied should show the domain, the mask and network" do
+    subnet = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8::', :name => 'subnet')
+    assert_equal "subnet (2001:db8::/64)", subnet.to_label
+  end
+
+  test "should find the subnet by ip" do
+    subnet = FactoryGirl.create(:subnet_ipv6)
+    assert_equal subnet, Subnet::Ipv6.subnet_for(get_ip(subnet, 10))
+  end
+
+  test "from cant be bigger than to range" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.to = get_ip(s, 10)
+    s.from = get_ip(s, 17)
+    refute s.valid?
+  end
+
+  test "should be able to save ranges" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.from = get_ip(s, 10)
+    s.to = get_ip(s, 17)
+    assert s.save
+  end
+
+  test "should not be able to save ranges if they dont belong to the subnet" do
+    s = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8:1::')
+    s.from = '2001:db8:2::1'
+    s.to = '2001:db8:2::2'
+    refute s.valid?
+  end
+
+  test "should not be able to save ranges if one of them is missing" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.from = get_ip(s, 10)
+    refute s.valid?
+    s.to = get_ip(s, 17)
+    assert s.valid?
+  end
+
+  test "should not be able to save ranges if one of them is invalid" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.from = get_ip(s, 10).gsub(/:[^:]*:/, ':xyz:')
+    s.to = get_ip(s, 17)
+    refute s.valid?
+  end
+
+  test "should strip whitespace before save" do
+    s = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8::')
+    s.network = " #{s.network}   "
+    s.mask = " #{s.mask}   "
+    s.gateway = " 2001:db8::1   "
+    s.dns_primary = " 2001:db8::2   "
+    s.dns_secondary = " 2001:db8::3   "
+    assert s.save
+    assert_equal '2001:db8::', s.network
+    assert_equal 'ffff:ffff:ffff:ffff::', s.mask
+    assert_equal '2001:db8::1', s.gateway
+    assert_equal '2001:db8::2', s.dns_primary
+    assert_equal '2001:db8::3', s.dns_secondary
+  end
+
+  test "should not allow an address greater than 45 characters" do
+    s = FactoryGirl.build(:subnet_ipv6, :mask => 9.times.map { 'abcd' }.join(':'))
+    refute s.valid?
+    assert_match /is invalid/, s.errors.full_messages.join("\n")
+  end
+
+  test "trailing colons in addresses are invalid" do
+    refute FactoryGirl.build(:subnet_ipv6, :network => '2001:db8::1:').valid?
+  end
+
+  test "more than 4 chars in a block is invalid" do
+    refute FactoryGirl.build(:subnet_ipv6, :network => '2001:db8:abcde::1').valid?
+  end
+
+  # test module StripWhitespace which strips leading and trailing whitespace on :name field
```

Comment:
Again, this is tested on ipv4, so move them both to the test for the parent class (it's where these things are happening)